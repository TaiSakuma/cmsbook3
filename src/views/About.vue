<template>
  <div class="g-container">
    <div class="text-h2">This is an about page</div>
  </div>
</template>

<script setup lang="ts"></script>

<style scoped>
.g-container {
  display: grid;
  height: 100%;
  place-items: center;
}
</style>
