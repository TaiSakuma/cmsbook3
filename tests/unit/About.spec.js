import Vue from "vue";
import VueRouter from "vue-router";
import Vuetify from "vuetify";
import { mount, createLocalVue } from "@vue/test-utils";

import About from "@/views/About.vue";

Vue.use(Vuetify);
Vue.use(VueRouter);

const localVue = createLocalVue();

describe("About.vue", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(About, {
      localVue,
    });
  });

  it("test text", () => {
    expect(wrapper.text()).toBe("This is an about page");
  });

  it("test html", () => {
    expect(wrapper.html()).toContain("<h1>This is an about page</h1>");
  });

  it("match snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
