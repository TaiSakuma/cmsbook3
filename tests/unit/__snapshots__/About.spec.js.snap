// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About.vue match snapshot 1`] = `
<div class="container container--fluid fill-height">
  <div class="row align-center justify-center">
    <h1>This is an about page</h1>
  </div>
</div>
`;
